odl/tomo/analytic/filtered_back_projection.py

```diff
@@ -21,14 +21,160 @@
+from odl.tomo.util.utility import perpendicular_vector
 
 
-__all__ = ('fbp_op',)
+__all__ = ('fbp_op', 'tam_danielson_window')
 
 
-def fbp_op(ray_trafo, padding=True):
+def _axis_in_detector(geometry):
+    """A vector in the detector plane that points along the rotation axis."""
+    du = geometry.det_init_axes[0]
+    dv = geometry.det_init_axes[1]
+    axis = geometry.axis
+    c = np.array([np.vdot(axis, du), np.vdot(axis, dv)])
+    cnorm = np.linalg.norm(c)
+
+    # Check for numerical errors
+    assert cnorm != 0
+
+    return c / cnorm
+
+
+def _rotation_direction_in_detector(geometry):
+    """A vector in the detector plane that points in the rotation direction."""
+    du = geometry.det_init_axes[0]
+    dv = geometry.det_init_axes[1]
+    axis = geometry.axis
+    det_normal = np.cross(du, dv)
+    rot_dir = np.cross(axis, det_normal)
+    c = np.array([np.vdot(rot_dir, du), np.vdot(rot_dir, dv)])
+    cnorm = np.linalg.norm(c)
+
+    # Check for numerical errors
+    assert cnorm != 0
+
+    return c / cnorm
+
+
+def _fbp_filter(norm_freq, filter_type, filter_cutoff):
+    """Create a smoothing filter for FBP.
+
+    Parameters
+    ----------
+    norm_freq : `array-like`
+        Normalized frequencies.
+    filter_type : {'Ram-Lak', 'Shepp-Logan', 'Cosine', 'Hamming', 'Hann'}
+        The type of filter to be used.
+    filter_cutoff : float
+        Relative cutoff frequency for the filter.
+
+    Returns
+    -------
+    smoothing_filter : `numpy.ndarray`
+
+    References
+    ----------
+    http://oftankonyv.reak.bme.hu/tiki-index.php?page=Reconstruction
+    """
+    if not (0 < filter_cutoff <= 1):
+        raise ValueError('`filter_cutoff` ({}) not in the interval (0, 1]'
+                         ''.format(filter_cutoff))
+
+    indicator = (norm_freq <= filter_cutoff)
+
+    if filter_type == 'Ram-Lak':
+        filt = indicator
+    elif filter_type == 'Shepp-Logan':
+        filt = indicator * np.sinc(norm_freq / (2 * filter_cutoff))
+    elif filter_type == 'Cosine':
+        filt = indicator * np.cos(norm_freq * np.pi / (2 * filter_cutoff))
+    elif filter_type == 'Hamming':
+        filt = indicator * (0.54 +
+                            0.46 * np.cos(norm_freq * np.pi / (filter_cutoff)))
+    elif filter_type == 'Hann':
+        filt = indicator * np.cos(norm_freq * np.pi / (2 * filter_cutoff)) ** 2
+    else:
+        raise ValueError('unknown `filter_type` ({})'
+                         ''.format(filter_type))
+
+    return filt
+
+
+def tam_danielson_window(ray_trafo, smoothing_width=0.05):
+    """Create Tam-Danielson window from a `RayTransform`.
+
+    The Tam-Danielson window is an indicator function on the minimal set of
+    data needed to reconstruct a given data. It is useful in analytic
+    reconstruction methods such as FBP to give a more accurate reconstruction.
+
+    Parameters
+    ----------
+    ray_trafo : `RayTransform`
+        The ray transform that the window should be computed for.
```

Comment:
Unnecessarily hard to digest due to the split logic ("for" at the very end). More compact:
"The ray transform for which the window should be computed."  or
"The ray transform for which to compute the window."